### PR TITLE
Also persist QoS 0 subscriptions when clean flag false

### PR DESCRIPTION
**Pull request by mitsos1os**

### CODE REVIEW

**Comment by mitsos1os** [comment]
@mcollina Please check what is going on with the Travis Build and it fails.. .Locally with Node v8.7.0 everything succeeds.. Thanks

**Comment by mitsos1os** [comment]
@mcollina Sorry to insist on the issue, but because we are using Mosca in a production environment, and this bug affects us, I would prefer to use the corrected version instead of monkey patching the module code after running, could you please check if anything else is needed for the Travis Build to work? Because I think the errors are irrelevant, since in my local environment all tests succeed.

File: lib/persistence/levelup.js [hunk 1]
```diff
@@ -162,11 +162,12 @@ LevelUpPersistence.prototype.storeSubscriptions = function(client, done) {
...
+    // Issue #693
     Object.keys(client.subscriptions).forEach(function(key) {
-      if (client.subscriptions[key].qos > 0) {
-        subscriptions[key] = client.subscriptions[key];
-        subscriptions[key].ttl = that.options.ttl.subscriptions + now;
-      }
+      //if (client.subscriptions[key].qos > 0) { // Issue #693
+      subscriptions[key] = client.subscriptions[key];
+      subscriptions[key].ttl = that.options.ttl.subscriptions + now;
+      //} // Issue #693
```
Comment (mcollina) [inline lib/persistence/levelup.js]: can you remove these comments?

File: lib/persistence/redis.js [hunk 2]
```diff
@@ -283,10 +283,11 @@ RedisPersistence.prototype.storeSubscriptions = function(client, cb) {
...
   var subscriptions = {};
 
+  // Issue #693
   Object.keys(client.subscriptions).forEach(function(key) {
-    if (client.subscriptions[key].qos > 0) {
-      subscriptions[key] = client.subscriptions[key];
-    }
+    // if (client.subscriptions[key].qos > 0) { // Issue #693
+    subscriptions[key] = client.subscriptions[key];
+    //} // Issue #693
```
Comment (mcollina) [inline lib/persistence/redis.js]: can you remove these comments?

File: test/persistence/abstract.js [hunk 3]
```diff
@@ -558,7 +558,7 @@ module.exports = function(create, buildOpts) {
       });
     });
 
-    it("should not store a QoS 0 subscription", function(done) {
+    it("should store a QoS 0 subscription", function(done) {
```
Comment (mcollina) [inline test/persistence/abstract.js]: can you please check/add a test that we are not enqueuing QoS 0 packets?

Comment (mitsos1os) [reply test/persistence/abstract.js]: Can you tell me what you mean? You already have test for QoS 0 packages

Comment (mcollina) [reply test/persistence/abstract.js]: forget my comment :)

**Comment by mitsos1os** [comment]
Ok just removed the comments! Thanks!

File: test/persistence/abstract.js (hunk 3 above)
Comment (mitsos1os) [reply test/persistence/abstract.js]: Eeeee... what comment???? ;)

**Comment by mcollina** [comment]
Thanks! Releasing asap.

**Pull request merged**